wordpress-mobile/WordPress-Android
# Feature/9182 post filters rows ui

Partially fixes #9182  

**Note**
- The target branch should be updated to "feature/master-post-filters" when #9430 is merged.
- This PR doesn't try to refactor the code into the optimal state, it's just another iteration towards that goal.

**Changes**
- Extracts logic related to items in the PostList into a separate class - `PostListItemUiStateHelper`
- Updates the ui of post list items according to the current design (The retry button UI doesn't correspond to Zeplin - updating Zeplin is in Sylvester's backlog)

**TODO**
- implement restore and permanent delete actions
- only one label (eg. Local Draft, Uploading post etc.) is displayed at a time. It's an easy fix, I'm just not sure which labels are compatible.
- write unit tests

**To test:**
This PR touches all logic related to UI of items in the post list
- which labels are displayed (local draft, uploading, errors...)
- which actions are displayed (View, Submit/Publish/Retry/.., Edit, Trash/Delete ..)
- feature image, title, excerpt
- progress indicator, disabled overlay

Make sure clicks on actions are tracked and the actions work (don't forget to check the actions under "More" are also being tracked).

I would suggest testing only a couple of basic use cases and test the rest when we have all the unit tests + all the logic is implemented.

**Update release notes:**
- We'll update the release not when we merge the working branch into develop
